File: scripts/generate-icons.js
#!/usr/bin/env node
/**
 * Generate PWA icons using Canvas
 * Run: node generate-icons.js
 */

const fs = require('fs');
const path = require('path');

// For now, let's create placeholders and a guide
const iconSizes = [72, 96, 128, 144, 152, 192, 384, 512];

console.log('\n=== Nous PWA Icon Generator ===\n');
console.log('To generate icons from your SVG logo, you have two options:\n');

console.log('OPTION 1: Use the HTML Generator (Recommended)');
console.log('1. Open generate-icons.html in your browser');
console.log('2. Click "Generate Icons" button');
console.log('3. Save all downloaded PNG files to the /icons folder\n');

console.log('OPTION 2: Manual Generation');
console.log('1. Visit https://realfavicongenerator.net/');
console.log('2. Upload your Nous logo SVG');
console.log('3. Generate and download all icon sizes');
console.log('4. Save them to the /icons folder with these names:\n');

iconSizes.forEach(size => {
    console.log(`   - icon-${size}x${size}.png`);
});

console.log('\nOPTION 3: Use your current SVG as favicon');
console.log('Your app will work with the current inline SVG favicon,');
console.log('but PNG icons are recommended for better compatibility.\n');

// Create placeholder icons directory structure
const iconsDir = path.join(__dirname, 'icons');
if (!fs.existsSync(iconsDir)) {
    fs.mkdirSync(iconsDir);
}

// Create a README in the icons folder
const readmeContent = `# App Icons

Place your generated PWA icons here with the following names:

${iconSizes.map(size => `- icon-${size}x${size}.png (${size}x${size} pixels)`).join('\n')}

## How to Generate Icons

### Method 1: Use generate-icons.html (Easiest)
1. Open generate-icons.html in your browser
2. Click "Generate Icons" button
3. Save all downloaded files here

### Method 2: Online Tool
1. Visit https://realfavicongenerator.net/
2. Upload your Nous logo
3. Generate and download icons
4. Save them here

### Method 3: Design Software
Use Figma, Sketch, or Photoshop to export your logo at these sizes.

## Your Logo SVG

Your current Nous logo is an inline SVG in index.html:
- Circle with 'n' text
- Font: Satoshi
- Color: #5d6b86
- Background: #f8f9fb
`;

fs.writeFileSync(path.join(iconsDir, 'README.md'), readmeContent);

console.log('✓ Created /icons folder with README');
console.log('✓ Ready for icon generation!\n');

File: scripts/generate-icons.py
#!/usr/bin/env python3
"""
Generate PWA icons from SVG logo
Requires: pip install cairosvg pillow
"""

import os
from io import BytesIO

try:
    import cairosvg
    from PIL import Image
except ImportError:
    print("Installing required packages...")
    os.system("pip3 install cairosvg pillow")
    import cairosvg
    from PIL import Image

# Your Nous logo SVG
svg_logo = """<?xml version="1.0" encoding="UTF-8"?>
<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 100 100'>
    <style>
        text {
            font-family: Arial, sans-serif;
            font-weight: 300;
            font-size: 72px;
            fill: #5d6b86;
        }
    </style>
    <rect width="100" height="100" fill="#f8f9fb"/>
    <circle cx='50' cy='50' r='35' fill='none' stroke='#000000' stroke-width='3'/>
    <text x='50' y='72' text-anchor='middle' letter-spacing='1.5'>n</text>
</svg>"""

# Icon sizes needed for PWA
sizes = [72, 96, 128, 144, 152, 192, 384, 512]

# Create icons directory if it doesn't exist
os.makedirs('icons', exist_ok=True)

print("Generating PWA icons from Nous logo...\n")

for size in sizes:
    try:
        # Convert SVG to PNG using cairosvg
        png_data = cairosvg.svg2png(
            bytestring=svg_logo.encode('utf-8'),
            output_width=size,
            output_height=size
        )

        # Save the PNG
        filename = f'icons/icon-{size}x{size}.png'
        with open(filename, 'wb') as f:
            f.write(png_data)

        print(f"✓ Generated {filename}")
    except Exception as e:
        print(f"✗ Failed to generate {size}x{size}: {e}")

print("\n✓ All icons generated successfully!")
print("Icons are saved in the /icons folder.")
